bitbots_ball_filter:
  ball_subscribe_topic: 
    type: string
    default_value: 'balls_relative'
    read_only: true
    description: 'Topic to subscribe to for ball detections'
  
  ball_position_publish_topic: 
    type: string
    default_value: 'ball_position_relative_filtered'
    read_only: true
    description: 'Topic to publish the filtered ball position'
  
  ball_filter_reset_service_name: 
    type: string
    default_value: 'ball_filter_reset'
    read_only: true
    description: 'Service to reset the ball filter'
  
  filter_frame: 
    type: string
    default_value: 'map'
    description: 'Frame to filter the ball in'
    validation:
      one_of<>: [['map', 'odom']]

  filter_rate: 
    type: int
    default_value: 62
    description: 'Filtering rate in Hz'
    read_only: true
    validation: 
      bounds<>: [0, 100]  
    
  process_noise_variance: 
    type: double
    default_value: 0.001
    description: 'Noise which is added to the estimated position without new measurements'
    validation: 
      bounds<>: [0.0, 1.0]
    
  measurement_certainty: 
    type: double
    default_value: 0.02
    description: 'Ball measurement certainty in filter'
    validation: 
      bounds<>: [0.0, 1.0]
  
  noise_increment_factor: 
    type: double
    default_value: 0.2
    description: 'Factor to increase the noise if the ball is further away. This also depends on the reference distance.'
    validation: 
      bounds<>: [0.0, 1.0]

  ignore_measurement_threshold: 
    type: double
    default_value: 3.0
    description: 'We calculate a distance around our filter state over which we ignore measurements. This threshold is based on the covariance of the filter scaled by this factor. This is used to ignore false positives if we already have a good estimation.'
    validation: 
      bounds<>: [0.0, 10.0]
